src/mscorlib/src/System/String.cs

```diff
@@ -1138,10 +1202,11 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
         //       sepList    -- an array of ints for split char indicies.
         //--------------------------------------------------------------------    
         [System.Security.SecuritySafeCritical]  // auto-generated
-        private unsafe int MakeSeparatorList(char[] separator, ref int[] sepList) {
+        private unsafe int MakeSeparatorList(char* separators, int separatorsLength, ref int[] sepList) {
+            Contract.Assert(separatorsLength >= 0, "separatorsLength >= 0");
             int foundCount=0;
 
-            if (separator == null || separator.Length ==0) {
+            if (separators == null || separatorsLength ==0) {
```

Comment:
Yeah, I think that's a good heuristic.  I really appreciate you defaulting to the thing that induces less churn.